### PR TITLE
Rewritten app/driver

### PR DESCRIPTION
**Pull request by robertklep**
* moved to SDK2
* updated dependencies
* added more capabilities (system pressure, outdoor temp, operating mode, clock programme state, central heating state, pressure alarm)
* added settings (update interval, pressure alarm settings, ability to change the password for the device when the hardware device's password got changed)
* more flowcards
* rewritten pairing FE

Squashed PR because I don't think you're interested in the history (if you are, I'll create a PR from the [`sdk2` branch](https://github.com/robertklep/nl.nefit.easy/tree/sdk2)).

Also, I'd be happy to take ownership of the app and keep it as a fork (instead of you merging this PR) or a new repo for the SDKv2 implementation (because there isn't much of the SDKv1-app left) under `https://github.com/robertklep`.

One Q: I'm not sure if SDKv2 apps are backward compatible with SDKv1 apps in terms of data/settings? Can users upgrade to the SDKv2 version and expect everything to work, or should the device be recreated on the Homey?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-const Log = require('homey-log').Log;
-
-module.exports.init = () => {
-	console.log(`${Homey.manifest.id} running...`);
-};
+module.exports = class NefitEasyApp extends Homey.App {
+  onInit() {
+    this.log(`${ Homey.manifest.id } is running...(debug mode ${ Homey.env.DEBUG ? 'on' : 'off' })`);
+    if (Homey.env.DEBUG) {
+      require('inspector').open(9229, '0.0.0.0');
```
Comment (RobinBol) [inline app.js]: This doesn't seem like you want it in a production app?

**Comment by robertklep** [comment]
> Regarding the migration, yes the app should be able to update from SDKv1 to SDKv2 otherwise it will break for all users when updating (only changes in capabilities will not work, then a user needs to re-add the device; in this case you should set the device unavailable with a message to re-add it). So it is important that you make sure the device_data object is the same as in the old app! (you can easily test if everything works by installing a nefit easy device with SDKv1 and then update the app to SDKv2.

Because of the many new capabilities, it doesn't work (`invalid_capability` errors). Making it unavailable showing a "re-add" message is a good idea.

> Besides, I see you made many new capabilities also for things as pressure and temperature measurements. Maybe it is useful to use the system capabilities for this? See apps.developer.athom.com/tutorial-Drivers-Capabilities.html 'Using the same capability more than once'.

I initially looked into that but wasn't able to get it working (also wasn't able to find any apps that use it, as an example).

As for pressure: the system capability has uses _mbar_ as unit, whereas I needed _bar_, hence the custom capability (I asked around on Slack but it seems there isn't a method to just override a unit for a system capability).

File: app.js (hunk 1 above)
Comment (robertklep) [reply app.js]: I figured that `Homey.env.DEBUG` would never evaluate to `true` in a regular install.

**Comment by RobinBol** [comment]
Do you want to push your changes for the setUnavailable when updating to the rewrite? Then I'll publish the app in beta in the app store, I am a bit hesitant to push it to stable since it forces our users to re-add their devices.

**Comment by robertklep** [comment]
Changes are in there ([here](https://github.com/athombv/nl.nefit.easy/pull/4/commits/9438bcf45b83e8dbe0d98e210eff4c19cbd23ff7#diff-756dab318a03f1eb776df4d3bafcf707R18)), I squashed them into the initial PR.

I would have suggested to run a beta test first myself, I'm the only user right now so who knows what other people may run in to :D

**Comment by RobinBol** [comment]
Ah I see, thanks. Well, I assumed you already did the testing on your own setup right?

**Comment by robertklep** [comment]
Absolutely, I've been running it for some time now and it works fine :)

However, I have a Nefit boiler, and I know other people combine the Easy with a non-Nefit boiler, which may yield different results.

**Comment by RobinBol** [comment]
Ok, well, I'll push it to the beta channel, it would be great if you could keep an eye on the comments/feedback! Maybe it is necessary to add a checkbox to the pairing wizard for people to select whether they have a Nefit boiler?

**Comment by robertklep** [comment]
From my own experience (during development of the `nefit-easy-*` modules), the likely feature that won't be supported with non-Nefit boilers is a pressure readout, which I already account for in the driver.

It's not that readouts will fail, the Easy will always return a value, it's just that the value may be bogus (in case of pressure, a value of 25.5 bar is returned, which is unreasonable high, so the driver accounts for that and doesn't set the capability value when it encounters that value).

**Comment by RobinBol** [comment]
Ah clear, great work, thanks for your contribution!

**Comment by robertklep** [comment]
Any timeline on when the beta is promoted to stable? It's been performing on my Homey without issues for the last 6 months.

**Comment by RobinBol** [comment]
I don't want to force the migration onto the stable users, so until we have a way to apply the migration in a non-breaking way I can't push it to stable.

**Pull request merged**